File: 557-reverse-words-in-a-string-iii/reverse-words-in-a-string-iii.cpp
class Solution {
// Helper method to swap characters in place
 void reverse(string& s, int left, int right) {
        while (left < right) {
            char temp = s[left];
            s[left] = s[right];
            s[right] = temp;
            left++;
            right--;
        }
    }

public:
    string reverseWords(string s) {
        
        int i =0;
        int j =0;
        
        while(i<s.size()){
            while(i<s.size() && s[i]!=' '){
                i++;
            }
            reverse(s,j,i-1);
            j=i+1;
            i++;

        }
        return s;
    }
    
};

File: 557-reverse-words-in-a-string-iii/reverse-words-in-a-string-iii.java
class Solution {
    public String reverseWords(String s) {
     char[] arr=s.toCharArray();
        int i =0;
        int j =0;
        
        while(i<arr.length){
            while(i<arr.length && arr[i]!=' '){
                i++;
            }
            reverse(arr,j,i-1);
            j=i+1;
            i++;

        }
        return new String(arr);
    }
    // Helper method to swap characters in place
    private void reverse(char[] arr, int left, int right) {
        while (left < right) {
            char temp = arr[left];
            arr[left] = arr[right];
            arr[right] = temp;
            left++;
            right--;
        }
}
}